src/PowerWAFInterface.cpp

```diff
@@ -99,11 +99,12 @@ extern "C"
 
         PowerWAF* waf   = reinterpret_cast<PowerWAF*>(handle);
         auto& addresses = waf->manifest.get_root_addresses();
-        *size           = addresses.size();
-        if (addresses.empty())
+        if (addresses.empty() || addresses.size() > UINT32_MAX)
         {
             return nullptr;
```

Comment:
Should also set `*size` to `0`, to be safest and consistent with the first early return.
Also, this being C++, `numeric_limits` can be used, e.g. `std::numeric_limits<uint32_t>::max()` or `std::numeric_limits<std::remove_pointer_t<decltype(size)>>::max()`